### ISSUE TITLE
FlxGame#new()'s confusing zoom parameter

### ISSUE THREAD

**Issue by JoeCreates** [HaxeFlixel/flixel#1977]
Following my change in #1889 to make scrollbounds account for zoom, a couple of people have had issues due to the zoom parameter of FlxGame#new() not being set to 1 (#1976).

It seems that in both cases there was confusion about the meaning of said parameter, where it was interpreted to have an actual impact on the scale of the game, whereas it's actual meaning now seems to be "what value of zoom should count as not zoomed". My change did not take this into account and so does not work for other values in here. I believe it could be fixed simply by dividing the value of zoom that is used in updateScrollBounds by the game's defaultZoom, although this wouldn't help with the continued misinterpreting of the scroll parameter, which seems to have lost it's utility since the introduction of scale modes.

In the coming weeks when I have time I'll modify updateScrollBounds to account for other FlxGame zoom values. Is there any reason for this parameter to be kept long term, or should it ultimately be removed with the next batch of breaking changes?
##

**Comment by Gama11** [comment]
Looks like there's actually not a single demo that has `Zoom != 1`. So, yeah, I doubt it's really necessary anymore with scale modes...

**Comment by Gama11** [comment]
Does the fix in 96dccf0 look sensible?

**Comment by JoeCreates** [comment]
Yeah, that looks correct. :)

**Comment by buckle2000** [comment]
I changed `Zoom` to 3 and nothing happened. Is that a bug?
Flixel 4.2.0

**Comment by Mouradif** [comment]
Same here, the Zoom parameter in FlxGame's constructor doesn't seem to have any effect.

**Comment by brettchalupa** [comment]
I've found a workaround that I hope this will help folks out!

It is definitely confusing to me that [the docs for `FlxGame`](https://api.haxeflixel.com/flixel/FlxGame.html) say the `Zoom` param zooms the entire game, "The default level of zoom for the game's cameras (e.g. 2 = all pixels are now drawn at 2x)." it does not actually seem to zoom the game.

Through trial and error, I've found that setting the `FlxG.scaleMode` to an instance of `FixedScaleAdjustSizeScaleMode` does the trick.

Example code:

``` haxe
addChild(new FlxGame(0, 0, MainMenuState, 2, 60, 60, true));
FlxG.scaleMode = new flixel.system.scaleModes.FixedScaleAdjustSizeScaleMode();
```

This, for me, is taking a `Project.xml` game with a width of 480 and a height of 320 and drawing it at 240x160, which is what I'm wanting.

The benefit to zooming this way instead of setting the `Width` and `Height` params of `FlxGame` is that it allows the game to take up the fullscreen resolution on mobile targets. It allows for an aspect ratio-agnostic entire-game zoom, which is what I want.

I think it'd be helpful to either remove the `Zoom` param or clarify in the docs what it does. Happy to help do the leg work, but I just need to know what's the right thing to do.
